language/vm/src/file_format.rs

```diff
@@ -278,13 +294,47 @@ pub struct FunctionHandle {
 #[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
 pub enum StructFieldInformation {
     Native,
-    Declared {
-        /// The number of fields in this type.
-        field_count: MemberCount,
-        /// The starting index for the fields of this type. `FieldDefinition`s for each type must
-        /// be consecutively stored in the `FieldDefinition` table.
-        fields: FieldDefinitionIndex,
-    },
+    Declared(Vec<FieldDefinition>),
+}
+
+//
+// Instantiations
+//
+// Instantiations point to a generic handle and its instantiation.
+// The instantiation can be partial.
+// So, for example, `S<T, W>`, `S<u8, bool>`, `S<T, u8>`, `S<X<T>, address>` are all
+// `StructInstantiation`s
+
+/// A complete or partial instantiation of a generic struct
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub struct StructDefInstantiation {
+    pub def: StructDefinitionIndex,
+    pub type_parameters: SignatureIndex,
```

Comment:
let's talk about this at a later time. 
I used this name as it was the one suggested in the PR that just had the binary format. So in a sense that was the name we agreed on.
However I am not sure I like it or if nothing else it is not consistent everywhere. 
My expectation is that we need few rename passes... and that is when we do any name changing

Reply:
Well I think per Sam & Todd's latest conversation (can't remember under which PR) we've decided to use parameters instead of formals and arguments instead of actuals. The reason why this one bothers me is that it maps "type_actuals" in your RFC PR to the opposite concept.

But yeah in general I agree with having the renaming passes.

Reply:
I agree but it turns out that on this PR it adds a lot of complexity. 
I went down that path for a bit and it costed me time because renaming was causing conflicts that did not help. And it added to the review, it brought in hundreds of line of code to the diff and more difficult to follow up.
I think we have a ton of renaming to do but let's keep renaming out of this diff